src/typechecker/derive-type.lisp

```diff
@@ -768,8 +767,15 @@ which have default instances. Currently only resolves Num :a -> Num Integer"
                                              (not (entail env expr-preds p)))
                                            (apply-substitution local-subs preds))))
 
+        ;; NOTE: This is where defaulting happens
+
+        (unless allow-deferred-predicates
+          (setf local-subs (compose-substitution-lists (default-subs env nil reduced-preds) local-subs))
+          (setf reduced-preds (apply-substitution local-subs reduced-preds))
+          (setf expr-types (apply-substitution local-subs expr-type)))
```

Comment:
Should be `expr-type`, `expr-types` is undefined here.

Reply:
Good catch. It looks like the assignment was unnecessary anyway.